### PR TITLE
Determine the CSV modifications to send to the build system

### PR DESCRIPTION
**Pull request by mprahl**
When Freshmaker rebuilds bundle images, it must modify some attributes in the ClusterServiceVersion (CSV) file of the bundle
image so that it is added to the upgrade graph appropriately in an index image.

Resolves CWFHEALTH-76

### CODE REVIEW

File: freshmaker/handlers/botas/botas_shipped_advisory.py [hunk 1]
```diff
@@ -259,6 +269,120 @@ def handle(self, event):
         db_event.transition(EventState.SKIPPED, msg)
         return []
 
+    @staticmethod
+    def _split_semver(version):
```
Comment (hluk) [inline freshmaker/handlers/botas/botas_shipped_advisory.py:273]: I guess this is unused / replaced by semver Python package, right?

Comment (mprahl) [reply freshmaker/handlers/botas/botas_shipped_advisory.py:273]: Yes, you are right. I forgot to remove it.

**Review by apaplaus — APPROVED** [review]
Seems fine for me, but it needs to be rebased as I've merged some changes and they might conflict with each other a little bit.

**Review by qixiang — APPROVED** [review]
LGTM

**Comment by mprahl** [comment]
@apaplaus and @qixiang the rebase changed the code in `handle` and `_prepare_builds`. Could you please take another look?

**Review by apaplaus — APPROVED** [review]

File: freshmaker/pyxis.py [hunk 2]
```diff
@@ -151,9 +152,8 @@ def get_latest_bundles(self, index_images):
         :rtype: list
         """
         # we need 'bundle_path_digest' to find ContainerImage of that bundle
-        include_fields = ['data.channel_name', 'data.version',
-                          'data.related_images', 'data.bundle_path_digest',
-                          'data.bundle_path']
+        include_fields = ['data.channel_name', 'data.version', 'data.related_images',
+                          'data.bundle_path_digest', 'data.bundle_path', 'data.csv_name']
```
Comment (qixiang) [inline freshmaker/pyxis.py:156]: I realized that `data.csv_name` doesn't exist in schema, maybe you were meant to get `data.csv_display_name`?

File: freshmaker/pyxis.py [hunk 3]
```diff
@@ -164,7 +164,13 @@ def get_latest_bundles(self, index_images):
 
             request_params['filter'] = \
                 f'latest_in_channel==true&source_index_container_path=={path}'
```
Comment (qixiang) [inline freshmaker/pyxis.py:166]: Should be `f'latest_in_channel==true and source_index_container_path=={path}'`, `&` is invalid logical operator in RSQL expression, it's not introduced in your PR, but could you also include a commit to fix this?

File: freshmaker/pyxis.py (hunk 2 above)
Comment (mprahl) [reply freshmaker/pyxis.py:156]: Sorry @qixiang, I forgot to provide context. This is going to be added to the schema in CWFHEALTH-285.

File: freshmaker/pyxis.py (hunk 3 above)
Comment (mprahl) [reply freshmaker/pyxis.py:166]: Good catch. Thank you.

**Comment by mprahl** [comment]
@qixiang could you please review again?

**Comment by qixiang** [comment]
LGTM

**Pull request merged**